packages/vaadin-grid/src/vaadin-grid-column.js

```diff
@@ -619,6 +517,207 @@ export const ColumnBaseMixin = (superClass) =>
       }
       this._previousHidden = hidden;
     }
+
+    /** @private */
+    __runRenderer(renderer, cell, model) {
+      const args = [cell._content, this];
+      if (model && model.item) {
+        args.push(model);
+      }
+
+      renderer.apply(this, args);
+    }
+
+    /**
+     * Renders either a template or a renderer to the given cells.
+     * Throws an error if both are passed, the template and the renderer.
+     *
+     * @private
+     */
+    __renderCellsContent(template, renderer, cells) {
+      // Skip if the column is hidden or not attached to a grid.
+      if (this.hidden || !this._grid) {
+        return;
+      }
+
+      if (template && renderer) {
+        throw new Error('You should only use either a renderer or a template');
+      }
+
+      cells.forEach((cell) => {
+        const model = this._grid.__getRowModel(cell.parentElement);
+
+        if (renderer) {
+          if (cell._renderer !== renderer) {
+            cell._content.innerHTML = '';
+          }
+
+          cell._renderer = renderer;
+
+          if (model.item || renderer === this.__headerRenderer || renderer === this.__footerRenderer) {
+            this.__runRenderer(renderer, cell, model);
+          }
+        } else if (cell._template !== template) {
+          cell._template = template;
+
+          cell._content.innerHTML = '';
+          template.templatizer._grid = template.templatizer._grid || this._grid;
+          const inst = template.templatizer.createInstance();
+          cell._content.appendChild(inst.root);
+          cell._instance = inst;
+          if (model.item) {
+            cell._instance.setProperties(model);
+          }
+        }
+      });
+    }
+
+    /**
+     * Renders the content of body cells using either a template or a renderer.
+     *
+     * @private
+     */
+    __renderBodyCellsContent() {
+      if (!this._cells) return;
+      if (!this._bodyTemplate && !this.__renderer) return;
+
+      this.__renderCellsContent(this._bodyTemplate, this.__renderer, this._cells);
+    }
+
+    /**
+     * Renders the header cell content using either a template or a renderer
+     * and then updates the visibility of the parent row depending on
+     * whether all its children cells are empty or not.
+     *
+     * @private
+     */
+    __renderHeaderCellContent() {
```

Comment:
Seems that many of the new observers don't follow the "Always include dependencies as observer arguments" -rule https://polymer-library.polymer-project.org/1.0/docs/devguide/observers#dependencies